consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -127,27 +120,37 @@ impl<T: Payload> ChainedBftSMR<T> {
     }
 
     /// Create a proposer election handler based on proposers
-    fn create_proposer_election(&self) -> Box<dyn ProposerElection<T> + Send + Sync> {
-        assert!(!self.proposers.is_empty());
+    fn create_proposer_election(
+        &self,
+        validators: &ValidatorVerifier,
+    ) -> Box<dyn ProposerElection<T> + Send + Sync> {
+        let proposers = validators.get_ordered_account_addresses();
         match self.config.proposer_type {
             ConsensusProposerType::MultipleOrderedProposers => {
-                Box::new(MultiProposer::new(self.proposers.clone(), 2))
+                Box::new(MultiProposer::new(proposers, 2))
```

Comment:
Unrelated to this PR, but we should probably explain the reason for 2 in more detail.